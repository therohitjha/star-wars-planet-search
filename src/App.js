import React, { useState, createContext } from "react";
import "./styles.css";
import { HashRouter, Switch, Route } from "react-router-dom";
import Home from "./Home";
import Login from "./Login";
export const LoginContext = createContext();
export default function App() {
  const [user, setUser] = useState({
    name: "Luke Skywalker",
    pass: "19BBY",
    isLogged: false,
    err: "",
  });

  const handleForm = (e) => {
    e.preventDefault();
    if (user.name !== "Luke Skywalker" || user.pass !== "19BBY") {
      setUser((prevData) => {
        return { ...prevData, err: "Wrong Username & Password" };
      });
    } else if (user.name === "Luke Skywalker" && user.pass === "19BBY") {
      setUser((prevData) => {
        return { ...prevData, isLogged: true };
      });
    }
  };

  return (
    <HashRouter basename="/">
      <LoginContext.Provider value={{ user, handleForm, setUser }}>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/login" component={Login} />
        </Switch>
      </LoginContext.Provider>
    </HashRouter>
  );
}
