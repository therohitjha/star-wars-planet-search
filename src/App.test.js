import { shallow } from "enzyme";
import React from "react";
import App from "./App";

  const user={
        name:'rohit',
        pass:'123'
    }
const handleForm=()=>{
    if(user.name==='rohit'){
        return 1
    }
}

describe("Testing app component", () => {
  it("App length testing", () => {
    expect.assertions(1);
    expect(shallow(<App />).length).toEqual(1);
  });

  it('taking snapshot',()=>{
      expect(handleForm()).toMatchSnapshot()
  })
  

  it("testing handleForm function", () => {
    expect(handleForm()).toEqual(1)
  });
});
