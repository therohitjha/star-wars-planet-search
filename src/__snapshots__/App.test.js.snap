// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing app component taking snapshot 1`] = `1`;
